subsys/net/ip/rpl.c

```diff
@@ -635,12 +635,16 @@ static void dio_timer(struct k_work *work)
 				net_if_ipv6_get_ll(rpl_default_iface,
 						   NET_ADDR_PREFERRED);
 
-			net_rpl_dio_send(rpl_default_iface, instance, addr,
-					 NULL);
-
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance, addr,
+					         NULL);
 #if defined(CONFIG_NET_STATISTICS_RPL)
 			instance->dio_send_pkt++;
 #endif
+			} else {
+				NET_DBG("Sending DIO failed, IPv6 link local "
+				        "address is NULL");
```

Comment:
Is it ok to resubmit with `instance->dio_next_delay` parameter in this case?

Reply:
IF addr is NULL should we?

`			k_delayed_work_submit(&instance->dio_timer,
					      DIO_TIMEOUT);
			return;`